sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```
@@ -238,10 +245,50 @@ case class Cast(child: Expression, dataType: DataType, timeZoneId: Option[String
   private lazy val dateFormatter = DateFormatter()
   private lazy val timestampFormatter = TimestampFormatter.getFractionFormatter(zoneId)
 
+  private lazy val customDateFormatter: DateFormatter = format.map { f =>
+    try {
+      DateFormatter(f)
+    } catch {
+      case NonFatal(_) => null
```

Is it better to fail here instead of silently ignore user's wrong pattern.

---

Ok fixed. But these formatters are lazy so only during execution is an exception raised and caught and the casted result will be null.
Should we catch the wrong pattern during analysis?

---

> But these formatters are lazy so only during execution is an exception raised 

I think it is ok.

> Should we catch the wrong pattern during analysis?

Java's DateTimeFormatter inside of `TimestampFormatter` is lazy too [see](https://github.com/apache/spark/blob/ef2d63bfb1859232d166320896957c0543c059cc/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/TimestampFormatter.scala#L50). To force the checking, you need to parse/format something. I don't think it is worth to do that here.